Python Study Group Gift II
Team D - Case Study Project #5
Title: Loan Eligibility Checker


Project Tasks
Task 1: Create the Customer Variables


In [52]:
customer_name = "Daniel Okafor"
customer_id = "CUST001"

age = 35
monthly_income = 350000
existing_monthly_debt = 70000
requested_loan_amount = 1000000

loan_term = 12
credit_score = 720

existing_loan = False
employment_status = "Employed"

Task 2: Validate Customer Information


In [53]:
name_validate = customer_name != ""
customer_id_validate = customer_id != ""
age_validate = age >= 18 and age <= 65
income_validate = monthly_income >= 150000
loan_amount_validate = requested_loan_amount > 0 and requested_loan_amount <= 5000000


print("Customer Name:", name_validate)
print("Customer ID:", customer_id_validate)
print("Age:", age_validate)
print("Monthly Income:", income_validate)
print("Loan Amount:", loan_amount_validate)

Customer Name: True
Customer ID: True
Age: True
Monthly Income: True
Loan Amount: True


Task 3: Validate Customer Age

In [54]:
age_validate = age >= 18 and age <= 65

print("Age Valid:", age_validate)

Age Valid: True


Task 4: Validate Monthly Income

In [55]:
income_validate = monthly_income >= 150000

if income_validate:
    print("Income Requirement: Passed")
else:
    print("Income Requirement: Failed")

Income Requirement: Passed


Task 5: Validate the Loan Amount

In [56]:
loan_amount_validate = requested_loan_amount > 0 and requested_loan_amount <= 5000000

print("Loan Amount Valid:", loan_amount_validate)

Loan Amount Valid: True


Task 6: Determine the Credit Score Category


In [57]:
if credit_score >= 750:
    credit_category = "Excellent"
elif credit_score >= 650:
    credit_category = "Good"
elif credit_score >= 550:
    credit_category = "Fair"
else:
    credit_category = "Poor"

print("Credit Score Category:", credit_category)

Credit Score Category: Good


Task 7: Calculate the Debt-to-Income Ratio


In [58]:
debt_to_income_ratio = (existing_monthly_debt / monthly_income) * 100

print("Debt-to-Income Ratio:", debt_to_income_ratio, "%")

# Checking for the 40% requirement
if debt_to_income_ratio <= 40:
    print("DTI Requirement: Passed")
else:
    print("DTI Requirement: Failed")

Debt-to-Income Ratio: 20.0 %
DTI Requirement: Passed


Task 8: Validate Employment Status


In [59]:
if employment_status == "Employed":
    employment_result = "Employed - Passed"
elif employment_status == "Self-Employed":
    employment_result = "Self-Employed - Passed"
elif employment_status == "Unemployed":
    employment_result = "Unemployed - Failed"
else:
    employment_result = "Invalid Employment Status"


print("Employment Status:", employment_result)

Employment Status: Employed - Passed


Task 9: Check Existing Loan Status

In [60]:
if existing_loan == False:
    print("Existing Loan: None")
else:
    print("Existing Loan: Active")

Existing Loan: None


Task 10: Validate Loan Term


In [61]:
if loan_term == 6:
    loan_term_result = "6 months"
elif loan_term == 12:
    loan_term_result = "12 months"
elif loan_term == 24:
    loan_term_result = "24 months"
else:
    loan_term_result = "Invalid Loan Term"

print("Loan Term:", loan_term_result)

Loan Term: 12 months


Task 11: Create a Loan Eligibility Function

In [62]:
def check_loan_eligibility(age, monthly_income, ebt_to_income_ratio, credit_score, existing_loan, employment_status, loan_amount, loan_term):
    age_validate = age >= 18 and age <= 65
    income_validate = monthly_income >= 150000
    dti_validate = debt_to_income_ratio <= 40
    credit_score_validate = credit_score >= 650
    existing_loan_validate = existing_loan == False
    employment_validate = employment_status != "Unemployed"
    loan_amount_validate = loan_amount > 0 and loan_amount <= 5000000
    loan_term_validate = loan_term in [6, 12, 24]
    
    if (age_validate and income_validate and dti_validate and credit_score_validate and existing_loan_validate
        and employment_validate and loan_amount_validate and loan_term_validate):
        return "Eligible for Loan"
    else:
        return "Not Eligible for Loan"

result = check_loan_eligibility(age, monthly_income, debt_to_income_ratio, credit_score, 
                                existing_loan, employment_status, requested_loan_amount, loan_term)

print(result)

Eligible for Loan


Task 12: Create a Debt-to-Income Function

In [63]:
def calculate_dti(monthly_income, existing_monthly_debt):
    debt_to_income_ratio = (existing_monthly_debt / monthly_income) * 100
    return debt_to_income_ratio

dti = calculate_dti(monthly_income, existing_monthly_debt)

print("Debt-to-Income Ratio:", dti, "%")

Debt-to-Income Ratio: 20.0 %


Task 13: Create a Loan Repayment Function


In [64]:
def calculate_monthly_payment(loan_amount, loan_term):
    monthly_payment = loan_amount / loan_term
    return monthly_payment

monthly_payment = calculate_monthly_payment(requested_loan_amount, loan_term)

print("Estimated Monthly Payment: ₦", monthly_payment)

Estimated Monthly Payment: ₦ 83333.33333333333


Task 14: Determine Overall Eligibility

In [65]:
if (age_validate and income_validate and loan_amount_validate and credit_score >= 650 and debt_to_income_ratio <= 40 
    and existing_loan == False and employment_status != "Unemployed" and loan_term in [6, 12, 24]):
    print("Loan Eligibility: Eligible")
else:
    print("Loan Eligibility: Not Eligible")

Loan Eligibility: Eligible


Task 15: Use a for Loop

In [66]:
applicants = [{"name": "Daniel Okafor", "age": 35, "monthly_income": 350000, "existing_monthly_debt": 70000, "loan_amount": 1000000, 
               "loan_term": 12, "credit_score": 720, "existing_loan": False, "employment_status": "Employed"},

#applicants 2
              {"name": "Grace Bello","age": 42, "monthly_income": 500000, "existing_monthly_debt": 100000, "loan_amount": 2000000,
               "loan_term": 24, "credit_score": 780, "existing_loan": False, "employment_status": "Employed"},

#applicants 3
              {"name": "Samuel Ade", "age": 27, "monthly_income": 120000, "existing_monthly_debt": 20000, "loan_amount": 500000,
               "loan_term": 12, "credit_score": 700, "existing_loan": False, "employment_status": "Employed"},

#applicants 4
              {"name": "Anita James", "age": 50, "monthly_income": 400000, "existing_monthly_debt": 200000, "loan_amount": 1500000,
               "loan_term": 12, "credit_score": 680, "existing_loan": True, "employment_status": "Employed"}]

for applicant in applicants:

    name = applicant["name"]
    age = applicant["age"]
    monthly_income = applicant["monthly_income"]
    existing_monthly_debt = applicant["existing_monthly_debt"]
    loan_amount = applicant["loan_amount"]
    loan_term = applicant["loan_term"]
    credit_score = applicant["credit_score"]
    existing_loan = applicant["existing_loan"]
    employment_status = applicant["employment_status"]

     # Age validation
    age_validate = age >= 18 and age <= 65

    # Income validation
    income_validate = monthly_income >= 150000

    # Loan amount validation
    loan_amount_validate = loan_amount > 0 and loan_amount <= 5000000

In [67]:
    # Credit score category
    if credit_score >= 750:
        credit_category = "Excellent"
    elif credit_score >= 650:
        credit_category = "Good"
    elif credit_score >= 550:
        credit_category = "Fair"
    else:
        credit_category = "Poor"

    # Credit score eligibility
    credit_score_validate = credit_score >= 650

    # Debt-to-income ratio
    dti = (existing_monthly_debt / monthly_income) * 100
    dti_validate = dti <= 40

    # Employment validation
    if employment_status == "Employed": 
        employment_validate = True
    elif employment_status == "Self-Employed":
        employment_validate = True
    else:
        employment_validate = False


In [68]:
    # Existing loan validation
    existing_loan_validate = existing_loan == False

    # Loan term validation
    loan_term_validate = loan_term in [6, 12, 24]

    # Overall eligibility
    if (age_validate and income_validate and loan_amount_validate and credit_score_validate and dti_validate
        and existing_loan_validate and employment_validate and loan_term_validate):
        eligibility = "Eligible"
    else:
        eligibility = "Not Eligible"

    # Monthly payment
    monthly_payment = loan_amount / loan_term

    print("=====================================")
    print("Applicant:", name)
    print("Age Valid:", age_validate)
    print("Income Eligible:", income_validate)
    print("Credit Score Category:", credit_category)
    print(f"Debt-to-Income Ratio: {dti:.2f}%")
    print("Employment Status:", employment_status)
    print("Existing Loan:", "Active" if existing_loan else "None")
    print("Loan Amount Valid:", loan_amount_validate)
    print("Loan Term Valid:", loan_term_validate)
    print("Loan Eligibility:", eligibility)
    print(f"Estimated Monthly Payment: ₦{monthly_payment:,.2f}")


Applicant: Anita James
Age Valid: True
Income Eligible: True
Credit Score Category: Good
Debt-to-Income Ratio: 50.00%
Employment Status: Employed
Existing Loan: Active
Loan Amount Valid: True
Loan Term Valid: True
Loan Eligibility: Not Eligible
Estimated Monthly Payment: ₦125,000.00


Task 16: Use a while Loop

In [69]:
# Task 16: Using While Loop to Validate Monthly Income
while True:
    monthly_income = float(input("Enter monthly income: ₦"))

    if monthly_income <= 0:
        print("Monthly income must be greater than ₦0.")
    else:
        print("Income accepted.")
        break

Enter monthly income: ₦ 40000


Income accepted.


Task 17: Use an Infinite While Loop

In [70]:
# Task 17: Create a controlled infinite loop using:
while True:
    answer = input("Do you want to check another loan application? ").lower()

    if answer == "yes":
        print("Continuing the program...")
    elif answer == "no":
        print("Exiting the program.")
        break
    else:
        print("Please enter yes or no.")

Do you want to check another loan application?  yes


Continuing the program...


Do you want to check another loan application?  n


Please enter yes or no.


Do you want to check another loan application?  no


Exiting the program.


Task 18: Understand Variable Scope

In [71]:
# Task 18: Understanding Variable Scope

# GLOBAL VARIABLE
# This variable is created outside the function.
# It can be accessed from anywhere in the program.
institution_name = "SmartBank"


def generate_report():

    # LOCAL VARIABLE
    # This variable is created inside the function.
    # It only exists inside this function.
    # It cannot be accessed directly outside the function.
    report_title = "Loan Eligibility Report"

    print(institution_name)
    print(report_title)


# Calling the function
generate_report()


# The global variable can be accessed outside the function.
print(institution_name)

# The local variable cannot be accessed here.
# The following line would cause an error:
# print(report_title)

SmartBank
Loan Eligibility Report
SmartBank


Task 19: Generate a Loan Eligibility Report

In [72]:
customer_name = "Daniel Okafor"
customer_id = "CUST001"

age = 35
monthly_income = 350000
existing_monthly_debt = 70000
requested_loan_amount = 1000000
loan_term = 12
credit_score = 720
existing_loan = False
employment_status = "Employed"

In [73]:
def calculate_dti(monthly_income, existing_monthly_debt):
    """Calculate the customer's debt-to-income ratio."""
    return (existing_monthly_debt / monthly_income) * 100


def calculate_monthly_payment(loan_amount, loan_term):
    """Calculate estimated monthly loan payment."""
    return loan_amount / loan_term

debt_to_income_ratio = calculate_dti(monthly_income, existing_monthly_debt)

monthly_payment = calculate_monthly_payment(requested_loan_amount, loan_term)

age_validate = age >= 18 and age <= 65

income_validate = monthly_income >= 150000

loan_amount_validate = (requested_loan_amount > 0 and requested_loan_amount <= 5000000)

credit_score_validate = credit_score >= 650

dti_validate = debt_to_income_ratio <= 40

existing_loan_validate= existing_loan == False

employment_validate = employment_status in ["Employed", "Self-Employed"]

loan_term_validate = loan_term in [6, 12, 24]

In [74]:
if credit_score >= 750:
    credit_category = "Excellent"
elif credit_score >= 650:
    credit_category = "Good"
elif credit_score >= 550:
    credit_category = "Fair"
else:
    credit_category = "Poor"

In [75]:
if (age_validate and income_validate and loan_amount_validate and credit_score_validate and dti_validate
    and existing_loan_validate and employment_validate and loan_term_validate):
    eligibility = "Eligible"
else:
    eligibility = "Not Eligible"

In [76]:
print("================================================")
print("           LOAN ELIGIBILITY REPORT")
print("================================================")

print()
print(f"Customer Name: {customer_name}")
print(f"Customer ID: {customer_id}")

print()
print(f"Age: {age}")
print(f"Monthly Income: ₦{monthly_income:,.0f}")
print(f"Existing Monthly Debt: ₦{existing_monthly_debt:,.0f}")

print()
print(f"Credit Score: {credit_score}")
print(f"Credit Category: {credit_category}")

print()
print(f"Loan Amount: ₦{requested_loan_amount:,.0f}")
print(f"Loan Term: {loan_term} Months")
print(f"Existing Loan: {existing_loan}")
print(f"Employment Status: {employment_status}")

print()
print("----------------------------------------------------")
print("              ELIGIBILITY CHECKS")
print("----------------------------------------------------")

print()
print("Age Requirement:", "Passed" if age_validate else "Failed")
print("Income Requirement:", "Passed" if income_validate else "Failed")
print("Loan Amount:", "Passed" if loan_amount_validate else "Failed")
print("Credit Score Requirement:", "Passed" if credit_score_validate else "Failed")
print("Debt-to-Income Requirement:", "Passed" if dti_validate else "Failed")
print("Existing Loan Requirement:", "Passed" if existing_loan_validate else "Failed")
print("Employment Requirement:", "Passed" if employment_validate else "Failed")
print("Loan Term Requirement:", "Passed" if loan_term_validate else "Failed")

print()
print(f"Debt-to-Income Ratio: {debt_to_income_ratio:.0f}%")

print()
print("-------------------------------------------------")
print("              LOAN CALCULATION")
print("-------------------------------------------------")

print()
print(f"Requested Loan: ₦{requested_loan_amount:,.0f}")
print(f"Loan Term: {loan_term} Months")
print(f"Estimated Monthly Payment: ₦{monthly_payment:,.2f}")

print()
print("-----------------------------------------------------")
print("                FINAL RESULT")
print("-----------------------------------------------------")

print()
print(f"Loan Eligibility: {eligibility}")

print()
print("=========================================================")

           LOAN ELIGIBILITY REPORT

Customer Name: Daniel Okafor
Customer ID: CUST001

Age: 35
Monthly Income: ₦350,000
Existing Monthly Debt: ₦70,000

Credit Score: 720
Credit Category: Good

Loan Amount: ₦1,000,000
Loan Term: 12 Months
Existing Loan: False
Employment Status: Employed

----------------------------------------------------
              ELIGIBILITY CHECKS
----------------------------------------------------

Age Requirement: Passed
Income Requirement: Passed
Loan Amount: Passed
Credit Score Requirement: Passed
Debt-to-Income Requirement: Passed
Existing Loan Requirement: Passed
Employment Requirement: Passed
Loan Term Requirement: Passed

Debt-to-Income Ratio: 20%

-------------------------------------------------
              LOAN CALCULATION
-------------------------------------------------

Requested Loan: ₦1,000,000
Loan Term: 12 Months
Estimated Monthly Payment: ₦83,333.33

-----------------------------------------------------
                FINAL RESULT
--------

BONUS CHALLENGE

In [77]:
def calculate_dti(monthly_income, existing_monthly_debt):
    dti = (existing_monthly_debt / monthly_income) * 100
    return dti

def calculate_monthly_payment(loan_amount, loan_term):
    monthly_payment = loan_amount / loan_term
    return monthly_payment


def get_credit_category(credit_score):

    if credit_score >= 750:
        return "Excellent"
    elif credit_score >= 650:
        return "Good"
    elif credit_score >= 550:
        return "Fair"
    else:
        return "Poor"


applicants = [{"name": "Daniel Okafor", "age": 35, "monthly_income": 350000, "existing_monthly_debt": 70000, "loan_amount": 1000000, 
               "loan_term": 12, "credit_score": 720, "existing_loan": False, "employment_status": "Employed"},

#applicants 2
              {"name": "Grace Bello", "age": 42, "monthly_income": 500000, "existing_monthly_debt": 100000, "loan_amount": 2000000,
               "loan_term": 24, "credit_score": 780, "existing_loan": False, "employment_status": "Employed"},
              
#applicants 3
              {"name": "Samuel Ade", "age": 27, "monthly_income": 120000, "existing_monthly_debt": 20000, "loan_amount": 500000,
                "loan_term": 12, "credit_score": 700, "existing_loan": False, "employment_status": "Employed"},

#applicants 4
              {"name": "Anita James", "age": 50, "monthly_income": 400000, "existing_monthly_debt": 200000,"loan_amount": 1500000,
               "loan_term": 12, "credit_score": 680, "existing_loan": True, "employment_status": "Employed"}]


eligible_count = 0
not_eligible_count = 0

highest_credit_score = 0
highest_credit_applicant = ""

highest_income = 0
highest_income_applicant = ""

highest_dti = 0
highest_dti_applicant = ""

lowest_dti = 0
lowest_dti_applicant = ""

highest_loan_amount = 0
highest_loan_applicant = ""

income_failed_applicant = ""
existing_loan_failed_applicant = ""

eligible_applicants = []

In [78]:
for applicant in applicants:

    name = applicant["name"]
    age = applicant["age"]
    monthly_income = applicant["monthly_income"]
    existing_monthly_debt = applicant["existing_monthly_debt"]
    loan_amount = applicant["loan_amount"]
    loan_term = applicant["loan_term"]
    credit_score = applicant["credit_score"]
    existing_loan = applicant["existing_loan"]
    employment_status = applicant["employment_status"]

In [79]:
    dti = calculate_dti(monthly_income, existing_monthly_debt)

    age_valid = age >= 18 and age <= 65

    income_valid = monthly_income >= 150000

    loan_amount_valid = (loan_amount > 0 and loan_amount <= 5000000)

    credit_score_valid = credit_score >= 650

    dti_valid = dti <= 40

    existing_loan_valid = existing_loan == False

    employment_valid = (employment_status == "Employed" or employment_status == "Self-Employed")

    loan_term_valid = (loan_term == 6 or loan_term == 12 or loan_term == 24)



In [80]:
    if (age_valid and income_valid and loan_amount_valid and credit_score_valid and dti_valid and existing_loan_valid
        and employment_valid and loan_term_valid):
        
        eligibility = "Eligible"
        eligible_count = eligible_count + 1

        # Storing eligible applicant
        eligible_applicants.append(applicant)

    else:
        eligibility = "Not Eligible"
        not_eligible_count = not_eligible_count + 1

In [81]:
# Finding Highest Credit Score
if credit_score > highest_credit_score:
        highest_credit_score = credit_score
        highest_credit_applicant = name


# Finding Highest Monthly Income

if monthly_income > highest_income:
        highest_income = monthly_income
        highest_income_applicant = name

# Finding Highest DTI

if dti > highest_dti:
        highest_dti = dti
        highest_dti_applicant = name

# Finding Lowest DTI

if lowest_dti == 0 or dti < lowest_dti:
        lowest_dti = dti
        lowest_dti_applicant = name


# Finding Highest Requested Loan

if loan_amount > highest_loan_amount:
        highest_loan_amount = loan_amount
        highest_loan_applicant = name

# Finding Applicant Who Failed Income Requirement
if income_valid == False:
        income_failed_applicant = name

# Finding Applicant Who Failed Because of Existing Loan

if existing_loan == True:
        existing_loan_failed_applicant = name


In [82]:
print("==============================================================")
print("             BONUS CHALLENGE RESULTS")
print("==============================================================")

print()

print("Number of Eligible Applicants:", eligible_count)

print("Number of Not Eligible Applicants:", not_eligible_count)

print()

print("Highest Credit Score:", highest_credit_applicant, "-", highest_credit_score)

print()

print("Highest Monthly Income:", highest_income_applicant, "- ₦", f"{highest_income:,.0f}")

print()

print("Highest Debt-to-Income Ratio:", highest_dti_applicant, "-", f"{highest_dti:.2f}%")

print()

print("Lowest Debt-to-Income Ratio:", lowest_dti_applicant, "-", f"{lowest_dti:.2f}%")

print()

print("Failed Income Requirement:", income_failed_applicant)

print()

print("Failed Because of Existing Loan:", existing_loan_failed_applicant)

print()

print("Highest Requested Loan Amount:", highest_loan_applicant, "- ₦", f"{highest_loan_amount:,.0f}")

print()

print("Estimated Monthly Payment for Eligible Applicants:")

for applicant in eligible_applicants:

    monthly_payment = calculate_monthly_payment(applicant["loan_amount"], applicant["loan_term"])

    print(applicant["name"], "- ₦", f"{monthly_payment:,.2f}")

print()

print("================================================================")

             BONUS CHALLENGE RESULTS

Number of Eligible Applicants: 0
Number of Not Eligible Applicants: 1

Highest Credit Score: Anita James - 680

Highest Monthly Income: Anita James - ₦ 400,000

Highest Debt-to-Income Ratio: Anita James - 50.00%

Lowest Debt-to-Income Ratio: Anita James - 50.00%

Failed Income Requirement: 

Failed Because of Existing Loan: Anita James

Highest Requested Loan Amount: Anita James - ₦ 1,500,000

Estimated Monthly Payment for Eligible Applicants:



• How many applicants are eligible? -===
2 — Daniel Okafor and Grace Bello

• How many applicants are not eligible?===
2 — Samuel Ade and Anita James

• Which applicant has the highest credit score?----
Grace Bello — 780

• Which applicant has the highest monthly income?===
Grace Bello — ₦500,000

• Which applicant has the highest debt-to-income ratio?---
Anita James — 50%
    
• Which applicant has the lowest debt-to-income ratio?---
Samuel Ade — 16.67%

• Which applicant failed the income requirement?----
Samuel Ade
    
• Which applicant failed because of an existing loan?-------
Anita James
    
• Which applicant has the highest requested loan amount?----
Grace Bello — ₦2,000,000
    
• What is the estimated monthly payment for each eligible applicant?-------
Daniel Okafor — ₦83,333.33 and Grace Bello — ₦83,333.33